### PR TITLE
Add trace file for "next dev"

### PR DESCRIPTION
**Pull request by timneutkens**
Same as the trace file for `next build` except that this one is rotated the moment 50MB is hit to ensure the file does not grow uncontrollably as `next dev` is a long-running process in many cases.

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| buildDuration | 15.3s | 15.5s | ⚠️ +212ms |
| buildDurationCached | 4s | 3.9s | -52ms |
| nodeModulesSize | 183 MB | 183 MB | ⚠️ +3.16 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.678 | 3.69 | ⚠️ +0.01 |
| / avg req/sec | 679.73 | 677.47 | ⚠️ -2.26 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.914 | 1.901 | -0.01 |
| /error-in-render avg req/sec | 1305.91 | 1315.18 | +9.27 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| 779.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 26.9 kB | 26.9 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 70.8 kB | 70.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| _app-HASH.js gzip | 977 B | 977 B | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 311 B | 311 B | ✓ |
| css-HASH.js gzip | 328 B | 328 B | ✓ |
| dynamic-HASH.js gzip | 2.67 kB | 2.67 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 918 B | 918 B | ✓ |
| image-HASH.js gzip | 4.15 kB | 4.15 kB | ✓ |
| index-HASH.js gzip | 260 B | 260 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 319 B | 319 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 13 kB | 13 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| _buildManifest.js gzip | 494 B | 494 B | ✓ |
| Overall change | 494 B | 494 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| index.html gzip | 539 B | 539 B | ✓ |
| link.html gzip | 551 B | 551 B | ✓ |
| withRouter.html gzip | 533 B | 533 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| buildDuration | 7.9s | 8.2s | ⚠️ +246ms |
| buildDurationCached | 3.8s | 3.8s | -68ms |
| nodeModulesSize | 183 MB | 183 MB | ⚠️ +3.16 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.539 | 3.615 | ⚠️ +0.08 |
| / avg req/sec | 706.32 | 691.5 | ⚠️ -14.82 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.091 | 1.867 | -0.22 |
| /error-in-render avg req/sec | 1195.4 | 1339.4 | +144 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| 675-HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| 770.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 50.7 kB | 50.7 kB | ✓ |
| main-HASH.js gzip | 34.9 kB | 34.9 kB | ✓ |
| webpack-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| Overall change | 101 kB | 101 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.33 kB | 1.33 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 315 B | 315 B | ✓ |
| css-HASH.js gzip | 331 B | 331 B | ✓ |
| dynamic-HASH.js gzip | 2.79 kB | 2.79 kB | ✓ |
| head-HASH.js gzip | 356 B | 356 B | ✓ |
| hooks-HASH.js gzip | 637 B | 637 B | ✓ |
| image-HASH.js gzip | 573 B | 573 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 2.2 kB | 2.2 kB | ✓ |
| routerDirect..HASH.js gzip | 326 B | 326 B | ✓ |
| script-HASH.js gzip | 393 B | 393 B | ✓ |
| withRouter-HASH.js gzip | 322 B | 322 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 10.1 kB | 10.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| _buildManifest.js gzip | 511 B | 511 B | ✓ |
| Overall change | 511 B | 511 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| index.html gzip | 539 B | 539 B | ✓ |
| link.html gzip | 551 B | 551 B | ✓ |
| withRouter.html gzip | 533 B | 533 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

</details>
Commit: 032e599e3c03a0edbe38f499567d0a7e6114718d

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 032e599e3c03a0edbe38f499567d0a7e6114718d

**test/integration/gssp-ssr-change-reloading/test/index.test.js** 
- GS(S)P Server-Side Change Reloading > should not reload page when client-side is changed too GSP
- GS(S)P Server-Side Change Reloading > should update page when getStaticProps is changed only
- GS(S)P Server-Side Change Reloading > should show indicator when re-fetching data
- GS(S)P Server-Side Change Reloading > should update page when getStaticPaths is changed only
- GS(S)P Server-Side Change Reloading > should not reload page when client-side is changed too GSSP
- GS(S)P Server-Side Change Reloading > should update page when getServerSideProps is changed only

<details>
<summary>Expand output</summary>

  ● GS(S)P Server-Side Change Reloading › should not reload page when client-side is changed too GSP

    Failed to replace content.

    Pattern: change me

    Content: import { useRouter } from 'next/router'

    export default function Gsp(props) {
      if (useRouter().isFallback) {
        return 'Loading...'
      }

      return (
        <>
          <p id="change">changed</p>
          <p id="props">{JSON.stringify(props)}</p>
        </>
      )
    }

    export const getStaticProps = async ({ params }) => {
      const count = 2

      if (params.post === 'second') {
        await new Promise((resolve) => setTimeout(resolve, 2000))
      }

      return {
        props: {
          count,
          params,
          random: Math.random(),
        },
      }
    }

    export const getStaticPaths = () => {
      /* eslint-disable-next-line no-unused-vars */
      const paths = 1

      return {
        paths: [{ params: { post: 'first' } }, { params: { post: 'second' } }],
        fallback: true,
      }
    }

      467 |     } else if (typeof pattern === 'string') {
      468 |       if (!currentContent.includes(pattern)) {
    > 469 |         throw new Error(
          |               ^
      470 |           `Failed to replace content.\n\nPattern: ${pattern}\n\nContent: ${currentContent}`
      471 |         )
      472 |       }

      at File.replace (lib/next-test-utils.js:469:15)
      at Object.<anonymous> (integration/gssp-ssr-change-reloading/test/index.test.js:49:63)

  ● GS(S)P Server-Side Change Reloading › should update page when getStaticProps is changed only

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 2

      71 |       async () =>
      72 |         JSON.parse(await browser.elementByCss('#props').text()).count + '',
    > 73 |       '1'
         |          ^
      74 |     )
      75 |   })
      76 |

      at Object.<anonymous> (integration/gssp-ssr-change-reloading/test/index.test.js:73:10)

  ● GS(S)P Server-Side Change Reloading › should show indicator when re-fetching data

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 2

      93 |     expect(await browser.eval(() => window.beforeChange)).toBe('hi')
      94 |     expect(await browser.eval(() => window.showedBuilder)).toBe(true)
    > 95 |     page.restore()
         |                 ^
      96 |
      97 |     await check(
      98 |       async () =>

      at Object.<anonymous> (integration/gssp-ssr-change-reloading/test/index.test.js:95:17)

  ● GS(S)P Server-Side Change Reloading › should update page when getStaticPaths is changed only

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 2

      87 |
      88 |     await check(
    > 89 |       async () =>
         |                  ^
      90 |         JSON.parse(await browser.elementByCss('#props').text()).count + '',
      91 |       '2'
      92 |     )

      at Object.<anonymous> (integration/gssp-ssr-change-reloading/test/index.test.js:89:29)

  ● GS(S)P Server-Side Change Reloading › should not reload page when client-side is changed too GSSP

    Failed to replace content.

    Pattern: change me

    Content: export default function Gssp(props) {
      return (
        <>
          <p id="change">changed</p>
          <p id="props">{JSON.stringify(props)}</p>
        </>
      )
    }

    export const getServerSideProps = ({ params }) => {
      const count = 2

      return {
        props: {
          count,
          params,
          random: Math.random(),
        },
      }
    }

      467 |     } else if (typeof pattern === 'string') {
      468 |       if (!currentContent.includes(pattern)) {
    > 469 |         throw new Error(
          |               ^
      470 |           `Failed to replace content.\n\nPattern: ${pattern}\n\nContent: ${currentContent}`
      471 |         )
      472 |       }

      at File.replace (lib/next-test-utils.js:469:15)
      at Object.<anonymous> (integration/gssp-ssr-change-reloading/test/index.test.js:163:63)

  ● GS(S)P Server-Side Change Reloading › should update page when getServerSideProps is changed only

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 2

      185 |       async () =>
      186 |         JSON.parse(await browser.elementByCss('#props').text()).count + '',
    > 187 |       '1'
          |          ^
      188 |     )
      189 |   })
      190 | })

      at Object.<anonymous> (integration/gssp-ssr-change-reloading/test/index.test.js:187:10)

</details>

File: packages/next/trace/report/to-json.ts [hunk 1]
```diff
@@ -3,11 +3,68 @@ import { batcher } from './to-zipkin'
...
+  sizeLimit: number
+  isRotating: Promise<void> | undefined
+  constructor(file: string, sizeLimit: number) {
+    this.file = file
+    this.size = 0
+    this.sizeLimit = sizeLimit
+    this.createWriteStream()
+  }
+  private createWriteStream() {
+    this.writeStream = fs.createWriteStream(this.file, writeStreamOptions)
```
Comment (sokra) [inline packages/next/trace/report/to-json.ts:29]: You need to read `this.size` from file here to initialize it when starting to append to an existing file.
I would also start by writing a `\n` to cancel any not completed line here.

**Review by sokra — CHANGES_REQUESTED** [review]

File: packages/next/trace/report/to-json.ts [hunk 2]
```diff
@@ -3,11 +3,68 @@ import { batcher } from './to-zipkin'
...
+
+    await new Promise<void>((resolve, reject) => {
+      this.writeStream.write(data, 'utf8', (err) => {
+        if (err) {
+          return reject(err)
+        }
+
+        resolve()
+      })
+    })
```
Comment (sokra) [inline packages/next/trace/report/to-json.ts:60]:
```suggestion
    if (!this.writeStream.write(data, 'utf8')) {
      await new Promise<void>((resolve, reject) => {
        this.writeStream.once('drain', resolve);
      })
    }
```

see https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback

**Review by sokra — CHANGES_REQUESTED** [review]

File: packages/next/trace/report/to-json.ts (hunk 1 above)
Comment (timneutkens) [reply packages/next/trace/report/to-json.ts:29]: The goal is to create a new file and not start appending to an existing one

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 80be5675a060b5ebd8382fae4c75680537468168

**test/integration/gssp-ssr-change-reloading/test/index.test.js** 
- GS(S)P Server-Side Change Reloading > should not reload page when client-side is changed too GSSP
- GS(S)P Server-Side Change Reloading > should update page when getServerSideProps is changed only

<details>
<summary>Expand output</summary>

  ● GS(S)P Server-Side Change Reloading › should not reload page when client-side is changed too GSSP

    Failed to replace content.

    Pattern: change me

    Content: export default function Gssp(props) {
      return (
        <>
          <p id="change">changed</p>
          <p id="props">{JSON.stringify(props)}</p>
        </>
      )
    }

    export const getServerSideProps = ({ params }) => {
      const count = 2

      return {
        props: {
          count,
          params,
          random: Math.random(),
        },
      }
    }

      467 |     } else if (typeof pattern === 'string') {
      468 |       if (!currentContent.includes(pattern)) {
    > 469 |         throw new Error(
          |               ^
      470 |           `Failed to replace content.\n\nPattern: ${pattern}\n\nContent: ${currentContent}`
      471 |         )
      472 |       }

      at File.replace (lib/next-test-utils.js:469:15)
      at Object.<anonymous> (integration/gssp-ssr-change-reloading/test/index.test.js:163:63)

  ● GS(S)P Server-Side Change Reloading › should update page when getServerSideProps is changed only

    expect(received).toBe(expected) // Object.is equality

    Expected: 1
    Received: 2

      185 |       async () =>
      186 |         JSON.parse(await browser.elementByCss('#props').text()).count + '',
    > 187 |       '1'
          |          ^
      188 |     )
      189 |   })
      190 | })

      at Object.<anonymous> (integration/gssp-ssr-change-reloading/test/index.test.js:187:10)

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| buildDuration | 17.3s | 18.2s | ⚠️ +945ms |
| buildDurationCached | 4.2s | 4.4s | ⚠️ +134ms |
| nodeModulesSize | 183 MB | 183 MB | ⚠️ +3.08 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.836 | 4.08 | ⚠️ +0.24 |
| / avg req/sec | 651.8 | 612.72 | ⚠️ -39.08 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.11 | 2.128 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1185.09 | 1174.89 | ⚠️ -10.2 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| 779.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 26.9 kB | 26.9 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 70.8 kB | 70.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| _app-HASH.js gzip | 977 B | 977 B | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 311 B | 311 B | ✓ |
| css-HASH.js gzip | 328 B | 328 B | ✓ |
| dynamic-HASH.js gzip | 2.67 kB | 2.67 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 918 B | 918 B | ✓ |
| image-HASH.js gzip | 4.15 kB | 4.15 kB | ✓ |
| index-HASH.js gzip | 260 B | 260 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 319 B | 319 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 13 kB | 13 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| _buildManifest.js gzip | 494 B | 494 B | ✓ |
| Overall change | 494 B | 494 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| index.html gzip | 539 B | 539 B | ✓ |
| link.html gzip | 551 B | 551 B | ✓ |
| withRouter.html gzip | 533 B | 533 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| buildDuration | 9.2s | 9.3s | ⚠️ +75ms |
| buildDurationCached | 4.5s | 4.4s | -140ms |
| nodeModulesSize | 183 MB | 183 MB | ⚠️ +3.08 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 4.114 | 4.256 | ⚠️ +0.14 |
| / avg req/sec | 607.72 | 587.47 | ⚠️ -20.25 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.208 | 2.158 | -0.05 |
| /error-in-render avg req/sec | 1132.37 | 1158.67 | +26.3 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| 675-HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| 770.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 50.7 kB | 50.7 kB | ✓ |
| main-HASH.js gzip | 34.9 kB | 34.9 kB | ✓ |
| webpack-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| Overall change | 101 kB | 101 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.33 kB | 1.33 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 315 B | 315 B | ✓ |
| css-HASH.js gzip | 331 B | 331 B | ✓ |
| dynamic-HASH.js gzip | 2.79 kB | 2.79 kB | ✓ |
| head-HASH.js gzip | 356 B | 356 B | ✓ |
| hooks-HASH.js gzip | 637 B | 637 B | ✓ |
| image-HASH.js gzip | 573 B | 573 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 2.2 kB | 2.2 kB | ✓ |
| routerDirect..HASH.js gzip | 326 B | 326 B | ✓ |
| script-HASH.js gzip | 393 B | 393 B | ✓ |
| withRouter-HASH.js gzip | 322 B | 322 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 10.1 kB | 10.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| _buildManifest.js gzip | 511 B | 511 B | ✓ |
| Overall change | 511 B | 511 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js add/trace-file-rotation | Change |
| - | - | - | - |
| index.html gzip | 539 B | 539 B | ✓ |
| link.html gzip | 551 B | 551 B | ✓ |
| withRouter.html gzip | 533 B | 533 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

</details>
Commit: 80be5675a060b5ebd8382fae4c75680537468168

**Pull request merged**